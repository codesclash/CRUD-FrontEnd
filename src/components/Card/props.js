import {
  deletePost,
  resetError,
  resetMessage,
} from "../../redux/actions/postActions";

const mapStateToProps = (state) => {
  return {
    successMessage: state.successMessage,
    error: state.error,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    deletePost: (id) => dispatch(deletePost(id)),
    resetError: () => dispatch(resetError()),
    resetMsg: () => dispatch(resetMessage()),
  };
};

export { mapStateToProps, mapDispatchToProps };
