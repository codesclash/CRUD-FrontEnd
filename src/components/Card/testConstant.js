export const post = {
  title: "title-title",
  content: "this is a test",
  author: "Deepak",
  datePublished: "2021-05-31T18:30:00.000Z",
  imgLink:
    "https://images.unsplash.com/photo-1619829950003-8b1b5e1b1f1a?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1050&q=80",
};
