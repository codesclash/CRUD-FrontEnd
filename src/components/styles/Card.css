.card-container{
    display: flex;
    flex-direction: column;
    position: relative;
    justify-content: flex-start;
    max-width: 330px;
    width: 330px;
    min-height: 450px;
    border-radius: 20px;
    background-color: #ffffff;
    box-shadow: 0 0 1rem rgba(0,0,0,0.2);
    transition: all 0.3s ease-in-out;
    overflow: hidden;
    margin-top: 1vh;
    word-break: break-word;
}
.card-container:hover{
    transform: translateY(-5px);
    box-shadow: 0 0 1rem rgba(0,0,0,0.5);
}
.card-img{
    width: 100%;
    max-height: 225px;
    min-height: 225px;
    position: relative;
}
.card-img > img{
    object-fit: cover;
    width: 100%;
    height: 100%;
}
.card-text{
    display: flex;
    flex-direction: column;
    justify-content: center;
    margin-top: 0.5rem;
    padding:1rem;
}

.card-title_des > h1{
    font-size: 1.5rem;
    font-weight: 700;
    color: #000;
    margin-bottom: 0.2rem;
}
.card-title_des > p{
    font-size: 1rem;
    font-weight: 400;
    color: #000;
    margin-bottom: 0.2rem;
}
.card-info{
    position: absolute;
    bottom: 70px;
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: flex-start;
    margin-top: 0.5rem;
    gap: 10px;
}
.card-btns{
    position: absolute;
    display: flex;
    flex-direction: row;
    justify-content: space-around;
    gap: 35px;
    bottom: 20px;
    margin-top: 1rem;
    
}
.card-btn{
    color: #ffffff;
    background-color: #8255D6;
    padding: 0.4rem 1rem;
    border-radius: 5px;
    font-weight: 700;
    font-size: 0.9rem;
    width: 75px;
    transition: all 0.3s ease-in-out;
    border: none;

}
.card-btn:hover{
    background-color: #6B3FA0;
    transform: scale(1.05);
}
.card-btn-del{
    color: #ffffff;
    background-color: #FF0000;
    padding: 0.4rem 1rem;
    border-radius: 5px;
    font-weight: 700;
    font-size: 0.9rem;
    transition: all 0.3s ease-in-out;
    border: none;
}
.card-btn-del:hover{
    background-color: #B30000;
    transform: scale(1.05);
}
.info{
    font-weight: 400;
    margin-top: 5px;
}

@media screen and (max-width: 768px) {
    .card-container{
        max-width: 300px;
        width: 300px;
        min-height: 400px;
    }
    .card-img{
        max-height: 200px;
        min-height: 200px;
    }
    .card-title_des > h1{
        font-size: 1.2rem;
    }
    .card-title_des > p{
        font-size: 0.9rem;
    }
    .card-btns{
        bottom: 16px;
    }
    .card-btn ,.card-btn-del {
        width: 60px;
        font-size: 12px;
        padding: 0.4rem 0.6rem;
    }
    .card-info{
        bottom: 60px;
        font-size: 0.8rem;
    }

    
}
@media screen and (max-width:425px) {
    .card-container{
        max-width: 290px;
        width: 280px;
        min-height: 400px;
    }
    .card-img{
        max-height: 180px;
        min-height: 180px;
    }
    .card-img > img{
        object-fit: cover;
        width: 100%;
        max-height: 180px;
    }
    .card-title_des > h1{
        font-size: 1.2rem;
    }
    .card-title_des > p{
        font-size: 0.8rem;
    }
    .card-btns{
        bottom: 16px;
    }
    .card-btn ,.card-btn-del {
        width: 60px;
        font-size: 12px;
        padding: 0.4rem 0.6rem;
    }
    .card-info{
        bottom: 70px;
        font-size: 0.8rem;
    }
    
    
}