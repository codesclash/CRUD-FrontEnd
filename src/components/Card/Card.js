import React, { Component } from "react";
import "../styles/Card.css";
import { Link } from "react-router-dom";
import "react-toastify/dist/ReactToastify.css";
import { ToastContainer} from "react-toastify";

//-----------------------------------------------------------MUI
import Button from "@mui/material/Button";
import { styled } from "@mui/material/styles";
import Dialog from "@mui/material/Dialog";
import DialogContent from "@mui/material/DialogContent";
import DialogActions from "@mui/material/DialogActions";
import Typography from "@mui/material/Typography";
import { createTheme,ThemeProvider } from '@mui/material/styles';
import purple from '@mui/material/colors/purple';

const theme = createTheme({
  palette: {
    primary: purple,
  },
});
const BootstrapDialog = styled(Dialog)(({ theme }) => ({
  "& .MuiDialogContent-root": {
    padding: theme.spacing(4),
  },
  "& .MuiDialogActions-root": {
    padding: theme.spacing(2),
  },
}));
 

 export default class Card extends Component {
  constructor(props) {
    super(props);
    this.state = {
      confirmDialog: false,
    };
  }

  componentDidMount() {
    console.log("abc")
  }


  handleClickOpen = () => {
    this.setState({
      confirmDialog: true,
    });
  };
  handleClose = () => {
    this.setState({
      confirmDialog: false,
    });
  };


  handleDelete = () => {
    this.props.deletePost(this.props.post._id);

    this.setState({
      confirmDialog: false,
    });

  };

  render() {

    const { title, content, author, datePublished, imgLink } = this.props.post;
    const slicedTitle =
      title.trim().length > 22
        ? title.trim().slice(0, 22).concat("...")
        : title;
    const slicedContent =
      content.trim().length > 70
        ? content.trim().slice(0,70).concat("...")
        : content;
    const slicedAuthor =
      author.trim().length > 25
        ? author.trim().slice(0,25).concat("...")
        : author;
    return (
      <>
      <div className="card-container">
        <div className="card-img">
          <img src={imgLink} alt="img" />
        </div>
        <div className="card-text">
          <div className="card-title_des">
            <h1>{slicedTitle}</h1>
            <p>{slicedContent}</p>
          </div>
          <div className="card-info">
            <h5>
              Author:<span className="info">{slicedAuthor}</span>
            </h5>
            <h5>
              Date Published:
              <span className="info">
                {new Date(datePublished).toLocaleDateString()}
              </span>
            </h5>
          </div>
          <div className="card-btns">
            <Link to={`/posts/${this.props.post._id}`}>
              <button className="card-btn">View</button>
            </Link>
            <Link to={`/post/update/${this.props.post._id}`}>
              <button className="card-btn">Edit</button>
            </Link>
            <button onClick={this.handleClickOpen} className="card-btn-del">
              Delete
            </button>
          </div>
        </div>
        <ThemeProvider theme = {theme}>
          <BootstrapDialog
            aria-labelledby="customized-dialog-title"
            open={this.state.confirmDialog}
          >
            <DialogContent  color="primary" dividers>
              <Typography gutterBottom>
                Are you sure you want to delete this post?
              </Typography>
            </DialogContent>
            <DialogActions >
              <Button onClick={this.handleClose}>Cancel</Button>
              <Button onClick={this.handleDelete} autoFocus>
                Delete
              </Button>
            </DialogActions>
          </BootstrapDialog>
          </ThemeProvider>
      </div>
      <ToastContainer />
      </>
    );
  }
}


