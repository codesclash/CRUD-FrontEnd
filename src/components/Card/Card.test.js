import React from "react";
import Card from "./Card";
import { post } from "./testConstant";
import {shallow,configure} from 'enzyme';
import Adapter from "@cfaester/enzyme-adapter-react-18";
configure({ adapter: new Adapter() });

describe("Test <Card />", () => {
  it("Should render Card component", () => {
    const component = shallow(<Card post={post} />);
    const tree = component.debug();
    expect(tree).toMatchSnapshot();
  });
  it("Title of the post should be title-title", () => {
    const component = shallow(<Card post={post} />);
 
    expect(component.find(".card-title_des").props().children[0].props.children).toBe(post.title);
  });

  it("Description of the post should be this is a test",()=>{
    const component = shallow(<Card post={post} />);

    expect(component.find(".card-title_des").props().children[1].props.children).toBe(post.content);
  })

  it("The name of the author should be Deepak",()=>{
    const component = shallow(<Card post={post} />);

    expect(component.find("#author").text()).toBe(post.author);

  })

  it("The date should be 2021-05-31T18:30:00.000Z",()=>{
    const component = shallow(<Card post={post} />);

    expect(component.find("#date").text()).toBe(new Date(post.datePublished).toLocaleDateString())});

  it("When clicking on delete button it should delete the post",()=>{
    const component = shallow(<Card post={post} />);
    const instance = component.instance();
   
    jest.spyOn(instance,"handleClickOpen");
    console.log(component.find("#del-btn").props())
    component.find("#del-btn").props().onClick();
   
    expect(component.state("confirmDialog")).toBe(true);
  });


});