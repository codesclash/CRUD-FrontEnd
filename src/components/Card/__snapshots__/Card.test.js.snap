// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Card /> Should render Card component 1`] = `
"<Fragment>
  <div className=\\"card-container\\">
    <div className=\\"card-img\\">
      <img src=\\"https://images.unsplash.com/photo-1619829950003-8b1b5e1b1f1a?ixid=MnwxMjA3fDB8MHxwaG90by1wYWdlfHx8fGVufDB8fHx8&ixlib=rb-1.2.1&auto=format&fit=crop&w=1050&q=80\\" alt=\\"img\\" />
    </div>
    <div className=\\"card-text\\">
      <div className=\\"card-title_des\\">
        <h1>
          title-title
        </h1>
        <p>
          this is a test
        </p>
      </div>
      <div className=\\"card-info\\">
        <h5>
          Author:
          <span className=\\"info\\" id=\\"author\\">
            Deepak
          </span>
        </h5>
        <h5>
          Date Published:
          <span className=\\"info\\" id=\\"date\\">
            6/1/2021
          </span>
        </h5>
      </div>
      <div className=\\"card-btns\\">
        <Link to=\\"/posts/undefined\\">
          <button className=\\"card-btn\\">
            View
          </button>
        </Link>
        <Link to=\\"/post/update/undefined\\">
          <button className=\\"card-btn\\">
            Edit
          </button>
        </Link>
        <button onClick={[Function (anonymous)]} className=\\"card-btn-del\\" id=\\"del-btn\\">
          Delete
        </button>
      </div>
    </div>
    <Styled(ForwardRef(Dialog)) aria-labelledby=\\"customized-dialog-title\\" open={false}>
      <ForwardRef(DialogContent) color=\\"primary\\" dividers={true}>
        <ForwardRef(Typography) gutterBottom={true}>
          Are you sure you want to delete this post?
        </ForwardRef(Typography)>
      </ForwardRef(DialogContent)>
      <ForwardRef(DialogActions)>
        <ForwardRef(Button) onClick={[Function (anonymous)]}>
          Cancel
        </ForwardRef(Button)>
        <ForwardRef(Button) onClick={[Function (anonymous)]} autoFocus={true}>
          Delete
        </ForwardRef(Button)>
      </ForwardRef(DialogActions)>
    </Styled(ForwardRef(Dialog))>
  </div>
  <ToastContainer position=\\"top-right\\" transition={[Function (anonymous)]} autoClose={5000} closeButton={[Function: E]} pauseOnHover={true} pauseOnFocusLoss={true} closeOnClick={true} draggable={true} draggablePercent={80} draggableDirection=\\"x\\" role=\\"alert\\" theme=\\"light\\" />
</Fragment>"
`;
